plugins/woocommerce/src/Admin/API/OnboardingPlugins.php

```diff
@@ -183,6 +195,38 @@ function( $action ) use ( $job_id ) {
 		return $response;
 	}
 
+
+	/**
+	 * Return Jetpack authorization URL.
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	public function get_jetpack_authorization_url() {
+		$manager = new Manager( 'woocommerce' );
+		// Register the site to wp.com.
+		if ( ! $manager->is_connected() ) {
+
+			$result = $manager->try_registration();
```

Comment:
It looks like `try_registration()` has a `$tos_agree` [param](https://github.com/Automattic/jetpack-connection/blob/218f75be81cc20d89307466cf258ada5a2fd8711/src/class-manager.php#L1228). I think we'll need to pass that state through here - with the plugin guidelines we would want to avoid Jetpack servers until users have knowingly agreed to it.

Reply:
On the plugins page, we have the following statement when you select Jetpack. 
![Screen Shot 2023-06-12 at 3 54 16 PM](https://github.com/woocommerce/woocommerce/assets/4723145/37f71e9a-a39c-4522-b38a-cd35af89c1a0)

Reply:
I think it could be solved with a `tos_agree` endpoint parameter as well. I think we need to consider that various parties will use this endpoint and there should be protection against accidental requests to Jetpack servers